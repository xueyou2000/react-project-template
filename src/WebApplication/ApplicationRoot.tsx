import Application from "@/Stores/Application";
import React, { useEffect, useRef } from "react";
import { RouteConfig } from "react-router-config";
import { Route, RouteComponentProps, Switch, withRouter } from "react-router-dom";
import { PageTransition } from "xy-page-transition";

export interface ApplicationRootProps extends RouteComponentProps {
    routes?: RouteConfig[];
}

function ApplicationRoot({ routes, location, history }: ApplicationRootProps) {
    const applicationRef = useRef(new Application(history));

    useEffect(() => {
        applicationRef.current.onRouterChange(location.pathname);
    }, [location.pathname]);

    return (
        <PageTransition timeout={300}>
            <Switch location={location}>
                {routes.map((config, i) => {
                    const RouteComponent = config.route || Route;
                    return <RouteComponent key={i} {...config} />;
                })}
            </Switch>
        </PageTransition>
    );
}

export default withRouter(ApplicationRoot);
