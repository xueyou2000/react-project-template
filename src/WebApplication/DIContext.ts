export const diContext: any = {};

/**
 * DI容器装饰器
 * @description 全局单例维护引用的容器
 * @example diInject() app;
 */
export function diInject(): any {
    return function(target: any, propertyKey: string) {
        Object.defineProperty(target, propertyKey, {
            get() {
                const bean = diContext[propertyKey];
                if (bean) {
                    return bean;
                } else if (process.env.NODE_ENV !== "production") {
                    console.error(`Context has no bean with name ${propertyKey}. 
                  Available beans: ${Object.getOwnPropertyNames(diContext).join(", ")}`);
                }
            },
            set() {
                throw new Error("Not allowed");
            },
            enumerable: true,
            configurable: true
        });
    };
}
