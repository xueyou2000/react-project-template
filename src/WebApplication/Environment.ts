const application = process.env.APPLICATION;
const baseurl = `/${application}`;

export interface Environment {
    /**
     * 项目基本标题
     */
    title: string;
    /**
     * 应用名
     */
    application: string;
    /**
     * 网关路径
     */
    baseurl: string;
    /**
     * 版本
     */
    version: string;
}

export default {
    baseurl,
    application,
    title: process.env.NAME,
    version: process.env.VERSION
} as Environment;
