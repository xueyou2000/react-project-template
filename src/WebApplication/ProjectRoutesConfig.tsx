import { RouteConfig } from 'react-router-config';
import LoadComponent from "@/Components/LoadComponent";

/**
 * 项目路由
 */
export const ProjectRoutesConfig: RouteConfig[] = [
    {
        title: '首页',
        path: "/welcome",
        component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
    },
    {
        title: '系统管理',
        path: "/system",
        routes: [
            {
                title: "菜单管理",
                path: "/system/menu",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/System/Menu")),
            },
            {
                title: "公众号配置",
                path: "/system/wechat",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
        ],
    },
    {
        title: '账户管理',
        path: "/account",
        routes: [
            {
                title: "账户查询",
                path: "/account/query",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
        ],
    },
    {
        title: '商户管理',
        path: "/customer",
        routes: [
            {
                title: "商户查询",
                path: "/customer/query",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
            {
                title: "商户开通",
                path: "/customer/create",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
        ],
    },
    {
        title: '订单管理',
        path: "/order",
        routes: [
            {
                title: "代付订单",
                path: "/order/dpay-order",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
            {
                title: "交易订单",
                path: "/order/trans-order",
                component: LoadComponent(() => import(/* webpackChunkName: "Test" */ "@/Pages/Test")),
            },
        ],
    },

];

export default ProjectRoutesConfig;