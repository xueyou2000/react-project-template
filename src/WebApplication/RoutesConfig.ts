import { RouteConfig } from "react-router-config";
import AuthorizedRoute from "@/Components/AuthorizedRoute";
import LoadComponent from "@/Components/LoadComponent";
import SystemLayout from "@/Layouts/SystemLayout";
import UserLayout from "@/Layouts/UserLayout";

export const RedirectPath = "/user/login";

export const ForgetPasswordtPath = "/user/forget-password";

export const ForgetPasswordtResultPath = `${ForgetPasswordtPath}-result`;

/**
 * 用户管理路由
 */
export const UserRoutesConfig: RouteConfig[] = [
    {
        path: RedirectPath,
        title: "用户登录",
        exact: true,
        component: LoadComponent(() => import(/* webpackChunkName: "user" */ "@/Pages/User/Login"))
    },
    {
        path: ForgetPasswordtPath,
        title: "找回密码",
        exact: true,
        component: LoadComponent(() => import(/* webpackChunkName: "user" */ "@/Pages/User/ResetPassword"))
    },
    {
        path: ForgetPasswordtResultPath,
        title: "找回密码完成",
        exact: true,
        component: LoadComponent(() => import(/* webpackChunkName: "user" */ "@/Pages/test"))
    }
];

/**
 * 根节点路由
 */
export const RootRoutesConfig: RouteConfig[] = [
    {
        path: "/",
        title: "运营后台",
        component: SystemLayout,
        exact: true,
        route: AuthorizedRoute
    },
    {
        path: "/user",
        title: "用户管理",
        component: UserLayout,
        routes: UserRoutesConfig
    },
    {
        path: "/",
        title: "运营后台",
        component: SystemLayout,
        route: AuthorizedRoute
    }
];
