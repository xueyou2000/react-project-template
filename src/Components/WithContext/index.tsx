import React from "react";
import { diContext } from "@/WebApplication/DIContext";

export default function withContext(Component: React.ComponentType<any>) {
    return (props: any) => <Component {...props} {...diContext} />;
}
