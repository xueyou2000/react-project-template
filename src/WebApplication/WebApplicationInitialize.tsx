import "@/Assets/Styles/base.scss";
import "@/Assets/Styles/index.scss";
import "@/Assets/Styles/global.scss";
import "@/Assets/Styles/overrite.scss";

import { RootRoutesConfig } from "@/WebApplication/RoutesConfig";
// import * as Sentry from "@sentry/browser";
import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import ApplicationRoot from "./ApplicationRoot";
import Environment from "./Environment";

// Sentry.init({ dsn: "https://6d7d4d0ae57b4518a3cda02f64cfe6b5@sentry.io/1483300" });

ReactDOM.render(
    <BrowserRouter basename={Environment.baseurl}>
        <ApplicationRoot routes={RootRoutesConfig} />
    </BrowserRouter>,
    document.getElementById("root")
);
