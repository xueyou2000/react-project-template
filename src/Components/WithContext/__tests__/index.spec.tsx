import React from "react";
import { render, fireEvent } from "@testing-library/react";
import WithContext from "../index";
import { diContext } from "@/WebApplication/DIContext";

function Profile({ name, age }) {
    return (
        <div>
            <p className="name">{name}</p>
            <p className="age">{age}</p>
        </div>
    );
}

const WithContextProfile = WithContext(Profile);

describe("WithContext", () => {
    test("auto injection context to props", () => {
        diContext.name = "XueYou";
        diContext.age = 16;
        const wrapper = render(<WithContextProfile />);

        const name = wrapper.container.querySelector(".name");
        const age = wrapper.container.querySelector(".age");
        expect(name.textContent).toBe("XueYou");
        expect(age.textContent).toBe("16");
        expect(wrapper).toMatchSnapshot();
    });
});
