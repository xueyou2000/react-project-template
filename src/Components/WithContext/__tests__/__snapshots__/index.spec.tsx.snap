// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithContext auto injection context to props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p
          class="name"
        >
          XueYou
        </p>
        <p
          class="age"
        >
          16
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p
        class="name"
      >
        XueYou
      </p>
      <p
        class="age"
      >
        16
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
